### PR TITLE
[WIP - do not merge!] code.org/teacher-dashboard redirects to studio.code.org/home

### PR DESCRIPTION
**Pull request by Erin007**
As part of the new site redesign, code.org/teacher-dashboard needs to redirect to studio.code.org/home 

NOTES: 
- studio.code.org/home doesn't exist yet, @breville will move all of the new homepage work to this url

### CODE REVIEW

**Comment by Erin007** [comment]
@tanyaparker @joshlory

**Comment by Bjvanminnen** [comment]
Presumably we want this behind a flag somewhere?

Do we really not need anything in pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml?

Worth checking whether paths like https://code.org/teacher-dashboard#/sections still work with this

**Comment by tanyaparker** [comment]
@Erin007 Totally forgot there are some hidden things on code.org/teacher-dashboard like the Ops Dashboard ([this stuff](https://github.com/code-dot-org/code-dot-org/blob/d344587164a1671898b827852c72e0e892918d79/pegasus/sites.v3/code.org/public/ops-dashboard/landing.haml)). We should make sure those are still accessible from your new teacher homepage. It can be totally ugly and look exactly the same since this only affects a small amount of super users. Just stick them at the bottom of the page.

**Comment by tanyaparker** [comment]
@Erin007 Have you tried CDO.studio_url('/home') for the relative redirect? I'm not sure if it can interpret Ruby in a .redirect file. And if there's no way to keep that relative, it's ok if that always redirects to production.

**Comment by Erin007** [comment]
Good call @brent - my first redirect attempt broke /sections etc. This new approach preserves all of that functionality.

@breville,  I check for the new_header cookie, and only redirect if it's there. 

@tanya, the Angular router doesn't seem to like CDO.studio_url 😞. I'll handle the Ops dashboard work in another PR.

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 1]
```diff
@@ -8,6 +8,9 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
 - error_string_none_selected = I18n.t('dashboard_error_none_selected').gsub(/"/,"\\\\\"")
 - error_string_other_section = I18n.t('dashboard_error_other_section').gsub(/"/,"\\\\\"")
 
+- if request.cookies['pm'] == 'new_header'
+  - homepageUrl = CDO.studio_url('/home')
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: Ruby variable should be `homepage_url`

File: pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml [hunk 2]
```diff
@@ -56,8 +59,15 @@ title: <%= I18n.t('dashboard_landing_title').inspect %>
   // ROUTES
 
   config(['$routeProvider', function($routeProvider) {
-    $routeProvider.when('/',
-        {templateUrl: '/teacher-dashboard/landing'});
+    if ('#{homepageUrl}') {
```
Comment (breville) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/index.haml]: Curious: what does this get set as, when `homepageUrl` isn't set?

**Comment by breville** [comment]
If well-tested, particular when the cookie isn't set, then just one change recommended to variable naming.

**Comment by breville** [comment]
Also you might want to change title before merging :)

**Pull request closed without merging**